src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7330,11 +7335,7 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
                         break;
 
                     default:
-                        resultState = NullableFlowState.MaybeNull;
-                        if (RequiresSafetyWarningWhenNullIntroduced(node.TargetType.TypeWithAnnotations))
-                        {
-                            ReportDiagnostic(ErrorCode.WRN_AsOperatorMayReturnNull, node.Syntax, type);
-                        }
+                        resultState = NullableFlowState.MaybeDefault;
```

Comment:
>MaybeDefault [](start = 56, length = 12)

Shouldn't we use this value only if the type is an unconstrained type parameter?